### PR TITLE
Fix two ITs for stability (PackagerIT.destroy() and RelationshipTypeRestControllerIT.findAllEntityTypes)

### LINKED ISSUE #8016: destroy() method in PackagerIT preventing *Builders cleanup
**Describe the bug**
The `destroy()` method in `PackagerIT` is used to remove a temp file and is overriding the parent method in _AbstractIntegrationTestWithDatabase._ As a result, *Builder objects are not cleaned up and this can affect other ITs. In my environment, this consistently causes `DiscoveryIT` to fail. Renaming the method  method in PackagerIT fixes the problem.

**To Reproduce**
It might be difficult to reproduce since others haven't seen the problem and CI tests succeed. Perhaps because of the order of test execution, I'm not sure. I see the error consistently when running `mvn clean install -DskipIntegrationTests=false -Dit.test=PackagerIT,DiscoveryIT -DfailIfNoTests=false`

**Expected behavior**
All ITs pass.

.

### PR DESCRIPTION
**Pull request by tdonohue**
## References
* Fixes #8016
* Also fixes a separate broken test in `RelationshipTypeRestControllerIT` which was not accounting for the "none" entity type. This test has caused recent PR builds to fail, including the first build of this PR: https://github.com/DSpace/DSpace/runs/4110898410?check_suite_focus=true

## Description
Fixes #8016 by ensuring the `PackagerIT.destroy()` method calls `super.destroy()`, so that the `destroy()` method in `AbstractIntegrationTestWithDatabase` is also triggered

Fixes separate test in `RelationshipTypeRestControllerIT`

## Instructions for Reviewers
See #8016 for instructions on how to reproduce the IT issue, which only exists if `PackagerIT` happens to run *before* `DiscoveryIT`.

### CODE REVIEW

**Review by mspalti — APPROVED** [review]
@tdonohue , I ran the tests locally and all passed.  So I can approve, with the minor suggestion for the `@Override`  annotation.

File: dspace-api/src/test/java/org/dspace/app/packager/PackagerIT.java
Comment (tdonohue) [reply dspace-api/src/test/java/org/dspace/app/packager/PackagerIT.java:89]: Good point! Added.

**Review by mspalti — APPROVED** [review]
Approving again . This is me experimenting with the platform. I'm not sure if this is necessary after your minor update.

**Comment by tdonohue** [comment]
@mspalti : Nope, it's not really necessary to re-approve after giving approval beforehand...*unless* you, as the reviewer, feel you want to re-express your approval.

Technically, it is possible to merge PRs quickly without _any approval_.  But, we don't tend to do that except in an emergency scenario (things badly broken & needing fixing on `main`).  So, as long as there is some prior "proof" of approval, that's all we really need to move a PR forward.

Thanks for your review here!  I'll wait for GitHub CI to finish it's build before merging.

**Pull request merged**
